config/logging.go

```diff
@@ -0,0 +1,41 @@
+package config
+
+// Logging configures overall and logger-specific log levels. Level values are
+// case-insensitive and include the following in order of importance: "fatal",
+// "panic", "dpanic", ""error", "warn", "info", "debug"
+type Logging struct {
+	// Level sets the log level for all loggers that do not have a setting in
+	// Loggers. The default value is "info".
+	Level string
+	// Loggers sets log levels for individual loggers.
+	Loggers map[string]string
+}
+
+// NewLogging returns Logging with values set to their defaults.
+func NewLogging() Logging {
+	return Logging{
+		Level: "info",
+		Loggers: map[string]string{
+			"basichost":    "warn",
+			"bootstrap":    "warn",
+			"dt_graphsync": "warn",
+			"dt-impl":      "warn",
+			"graphsync":    "warn",
```

Comment:
It might be more flexible to expose regex-based log level configuration and set it using `SetLogLevelRegex` instead of `SetLogLevel`.

Reply:
My preference is to match exactly. If the user does not know all the possible logger names, then a regex could unintentionally match a name and some important message would be discarded.

Reply:
The downside is that people whom know exactly what they want the logging to do would be forced to list subsystems individually which could become painful for core libraries like libp2p with multiple sub packages. 

We can certainly leave it as is and change when needed. This is by no means a blocker.